pilot/pkg/serviceregistry/kube/controller/ambientindex_external.go

```diff
@@ -562,13 +563,13 @@ func (a *AmbientIndexImpl) cleanupOldWorkloadEntriesInlinedOnServiceEntry(svcEnt
 	if oldServiceEntry, f := a.servicesMap[nsName]; f {
 		for _, oldWe := range oldServiceEntry.Spec.Endpoints {
 			oldUID := c.generateServiceEntryUID(nsName.Namespace, nsName.Name, oldWe.Address)
-			we, found := a.byUID[oldUID]
-			if found {
+			if we, found := a.byUID[oldUID]; found {
 				updates.Insert(model.ConfigKey{Kind: kind.Address, Name: we.ResourceName()})
 				for _, networkAddr := range networkAddressFromWorkload(we) {
 					delete(a.byWorkloadEntry, networkAddr)
 				}
 				delete(a.byUID, oldUID)
+				delete(a.byUID, c.generateWorkloadEntryUID(oldServiceEntry.GetNamespace(), oldServiceEntry.GetName()))
```

Comment:
It seems we donot store by svcEntry ns/name

Reply:
We store the inlined WE in https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller/authorization.go#L497

Inlined WE don't have their own name/ns so they "inherit" the embedding SE name/ns.